pkg/kubelet/kubelet_volumes.go

```diff
@@ -127,8 +129,23 @@ func (kl *Kubelet) cleanupOrphanedPodDirs(pods []*v1.Pod, runningPods []*kubecon
 			continue
 		}
 		if len(volumePaths) > 0 {
-			orphanVolumeErrors = append(orphanVolumeErrors, fmt.Errorf("Orphaned pod %q found, but volume paths are still present on disk", uid))
-			continue
+			for _, volumePath := range volumePaths {
+				// IsNotLikelyMountPoint() not good enough here, volume mounts are frequently bind-mounted, also not on a hot path
+				isNotMounted, err := kl.mounter.IsNotMountPoint(volumePath)
+				if err != nil {
+					orphanVolumeErrors = append(orphanVolumeErrors, fmt.Errorf("Orphaned pod %q found, but unable to evaluate mountpoints: %v", uid, err))
+					continue PodLoop
+				}
+				if isNotMounted {
+					if err := os.Remove(volumePath); err != nil { // Attempt to remove volume, will fail if there is for example data in there or a permission problem
```

Comment:
if `volumePath` is also used in a subpath and subpath is not cleaned up yet, removing the parent directory first will cause subpath to be never cleaned up.

```
~> mkdir -p foo bind_foo
~> mount -o bind foo bind_foo
~> rm -rf foo
~> rm -rf bind_foo
rm: cannot remove 'bind_foo': Device or resource busy
````

cc @msau42

Reply:
Now that I think of, if subpath is unmounted we should be fine. But since we aren't cleaning volumes if it has subpaths(bind mounts) left on the disks, this change seems incompatible with that.

Reply:
This is mostly in there as a safety measure since the Remove command will fail if there are any files  left in there. It's not actually needed to remove anything.

Reply:
The problem isn't that `Remove` function will fail. Problem is - something may have bind mounted this directory via a subpath to somewhere else and `isNotMounted` only checks if this directory is a mount point.